src/mapgen_functions.cpp

```diff
@@ -1211,69 +1211,71 @@ void mapgen_subway( mapgendata &dat )
+                                            "################CC......\n"
+                                            "#################CCCCCCC",
+                                            mapf::ter_bind( ". # C ^ D",
+                                                    t_thconc_floor,
                                                     t_rock,
+                                                    t_strconc_wall,
                                                     t_railroad_rubble,
                                                     t_railroad_track_d ),
-                                            mapf::furn_bind( ". # ^ D",
+                                            mapf::furn_bind( ". # C ^ D",
+                                                    f_null,
                                                     f_null,
                                                     f_null,
                                                     f_null,
                                                     f_null ) );
             } else { // normal subway drawing
                 mapf::formatted_set_simple( m, point_zero,
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...^X^^^X^....^X^^^X^...\n"
-                                            "...-x---x-....-x---x-...\n"
-                                            "...^X^^^X^....^X^^^X^...",
+                                            "#..^X^^^X^....^X^^^X^..#\n"
```

Comment:
@Venera3 so are these sections undrivable now?  I'm try to make sure I understand.

Reply:
In practice (and possibly in theory as well), yes, assuming I've understood the changes correctly. There is no problem driving a single tile width vehicle straight ahead in a two tile wide corridor. However, there isn't enough room to turn one that's 3 tiles long (a normal foldable bicycle with a basket for "stuff") because there isn't enough room to turn a vehicle moving at a speed higher than the minimum one as there's zero wriggling room: you have to arrive in exactly the correct line and turn at exactly the correct tile (you'll crash into the wall moving faster than 1 tile per turn). You can't drive between the pairs of tracks, because you can't turn (there isn't enough room, as the space narrows down to two tiles when the angle is an odd multiple of 45 degrees. The change essentially mirrors that situation at the edges, with the difference that the narrow sections are the straight ones, not the angled ones).

I would expect a rail vehicle to run into the same issue, i.e. running into the wall when turning because there isn't enough room for the overshoot.